Sources/UIComponent/Components/View/ObservationBoundaryComponent.swift

```diff
@@ -0,0 +1,61 @@
+//  Created by Alex Little on 01/02/2025.
+
+import UIKit
+import IssueReporting
+
+/// Protects the ancestor or sibling views from reloading when model properties change that are accessed in this subheirarchy
+/// This can isolate heavy model updates to a single component without reloading an entire screen
+public struct ObservationBoundaryComponent<C: ComponentBuilder>: ComponentBuilder {
+    let componentBuilder: C
+
+    public init(component: C) {
+        self.componentBuilder = component
+    }
+
+    public func build() -> some Component {
+        ViewComponent<UIView>()
+            .with(\.componentEngine.component, componentBuilder)
+#if DEBUG
+            .constraint { c in
+                if c.maxSize.height == .infinity {
+                    reportIssue("You must provide a height for \(String(describing: componentBuilder)) in a ObservationBoundaryComponent's")
+                }
+
+                if c.maxSize.width == .infinity {
+                    reportIssue("You must provide a width for \(String(describing: componentBuilder)) in a ObservationBoundaryComponent's")
+                }
+                return c
+            }
+#endif
+            .eraseToAnyComponentOfView()
```

Comment:
if you forget to apply a fixed size to observable components, a runtime warning will be triggered